### DISCUSSION TITLE
String interpolation: commas as thousands separators

### DISCUSSION THREAD

**Discussion by ghost** [vlang/v#11878] [Ideas]
`println('${1000}')`  ->   1,000

frequently used

**Comment by larpon** [comment]
That's locale independent. In some places of the world that would be float value of one 🙂 so in this case the number would be off by 999 😬

**Comment by JalonSolov** [comment]
Yes, the separator would need to be locale dependent.

**Comment by ghost** [comment]
I'll be happy if I can use it just like java.

```java
int price = 1000000;
System.out.println(String.format("%,d", price));
```

**Comment by dumblob** [comment]
That's a big question - whether a locale-specific behavior shall be part of the language itself.

I myself would definitely say **no** in the context of *V* (because locales are a moving target which doesn't fit the purpose of *V* unlike e.g. some string matching programming languages etc.). But others might have a different opinion.

Btw. I don't think the Java example is a good one. IMHO string interpolation is, well, just interpolation. Unlike formatting. Interpolation has pretty limited **finite** scope whereas formatting has pretty much endless space of what to support and/or add.

**Comment by ghost** [comment]
BTW, V doesn't have Formatter?
Then why not have it?

**Comment by JalonSolov** [comment]
V does have a formatter... <https://github.com/vlang/v/blob/master/doc/docs.md#string-interpolation>

It just doesn't support "thousands separators".
